spec/function.dd

```diff
@@ -561,6 +561,8 @@ $(H2 $(LNAME2 property-functions, Property Functions))
     $(LI $(D @property) functions cannot be overloaded with non-$(D @property) functions with the same name.)
     $(LI $(D @property) functions can only have zero, one or two parameters.)
     $(LI $(D @property) functions cannot have variadic parameters.)
+    $(LI Binary assignment operators can be used if the appropriate getter and setter $(D @property) functions
+    exist.)
```

Comment:
> somewhere you should mention (and test in the other PR) the interplay with opBinary overloading

I'm assuming you mean the `opOpAssign` overloads.  Please correct me if I'm wrong.

The only ambiguity I see with the proposed implementation is when there is an `alias this`ed property and an binary assignment operator overload (i.e. `opOpAssign`).

As an illustration of how things work today, consider the following code where there is an `alias this`ed property `x`, and the assignment operator overload `opAssign`.

```D
import std.stdio;

struct TInt
{
    int mX;

    @property int x() { return mX; }
    @property void x(int v) { mX = v; }

    int mB;
    void opAssign(int v) { mB = v; }

    alias x this;
}

void main()
{
    TInt t;
    t.x = 4;
    t = 5;    // Which do you think should be called? `x(5)` or `opAssign(5)`
	
    assert(t.mX == 4);
    assert(t.mB == 5);
}
```
This behavior is already defined in the spec at https://dlang.org/spec/class.html#alias-this

> If the member is a class or struct, undefined lookups will be forwarded to the AliasThis member. 

Since `opAssign` is not undefined `t = 5` will call `t.opAssign(5)`, and the binary assignment operators (i.e. `opOpAssign`) should work in the same way.

Since this behavior is already defined in the spec, I propose we don't duplicate it here in the `Property Functions` section.

> and test in the other PR

I will be adding tests to the DMD PR to check that the implementation conforms to this specified behavior.  Thank you for bringing it to my attention.